src/llm/language_model/legacy/servable_initializer.cpp

```diff
@@ -114,6 +114,12 @@ Status LegacyServableInitializer::initialize(std::shared_ptr<GenAiServable>& ser
     }
     properties->bestOfLimit = nodeOptions.best_of_limit();
     properties->maxModelLength = parseMaxModelLength(parsedModelsPath);
+
+    if (nodeOptions.has_enable_tool_guided_generation()) {
+        properties->enableToolGuidedGeneration = nodeOptions.enable_tool_guided_generation();
+    } else {
+        properties->enableToolGuidedGeneration = false;
+    }
```

Comment:
```suggestion
    properties->enableToolGuidedGeneration = nodeOptions.enable_tool_guided_generation();
```

nit picking, but i think this is possible since this field has default value
leaving it as is creates 2 sources of truth: default value here and one in .proto